base_layer/core/src/base_node/state_machine_service/states/block_sync.rs

```diff
@@ -322,6 +325,25 @@ async fn synchronize_blocks<B: BlockchainBackend + 'static>(
                 info.tip_height = network_tip_height;
             }
 
+            // Searching through the orphan database for every block to be added during a large block sync is not
+            // efficient at the moment
+            if network_tip_height as i64 - sync_height as i64 > MISSING_BLOCKS_THRESHOLD as i64 {
```

Comment:
I don't think this is the correct place for this. 

If the large block sync was efficient, where would you do this? In other words, if that problem did not exist, would you still add this code? Let's rather add it in the correct place, if it's needed and fix the large block sync efficiency. Otherwise we end up with 2 pieces of code that need to be cleaned up later

Reply:
I have updated the PR description to motivate why this test and cleanup is in the correct place. It is also now a user-configurable option.